ext/pgsql/pgsql.c

```diff
@@ -4510,25 +4510,23 @@ static int php_pgsql_convert_match(const zend_string *str, const char *regex , s
 /* {{{ php_pgsql_add_quote
  * add quotes around string.
  */
-static int php_pgsql_add_quotes(zval *src, bool should_free)
+static zend_string* php_pgsql_add_quotes(zend_string *src)
 {
 	smart_str str = {0};
-
-	assert(Z_TYPE_P(src) == IS_STRING);
-	assert(should_free == 1 || should_free == 0);
+	zend_string *result;
 
 	smart_str_appendc(&str, 'E');
 	smart_str_appendc(&str, '\'');
-	smart_str_appendl(&str, Z_STRVAL_P(src), Z_STRLEN_P(src));
+	smart_str_append(&str, src);
 	smart_str_appendc(&str, '\'');
 	smart_str_0(&str);
 
-	if (should_free) {
-		zval_ptr_dtor(src);
-	}
-	ZVAL_NEW_STR(src, str.s);
+	result = zend_string_dup(str.s, false);
```

Comment:
Why duplicate the string here? Maybe you want to just shrink to fit?

Reply:
I was hitting a memory leak and I thought it stemmed from there, might be incorrect I'll revisit.